### PR TITLE
Fixed Toolbar UI

### LINKED ISSUE #5118: a11y - expand/shrink toolbar - cms ui
**Describe the bug**
The button called "toolbar-handler", locate inside the toolbar, have a fixed `aria-label="Shrink toolbar"` even having an expand/shrink behavior.

**To Reproduce**

1. Go, using a screen reader and shift + tab keys, to the button that controls the toolbar
2. Press enter, the toolbar will shrink
3. Another press, the toolbar will expand.

**Expected behavior**
The screen-reader needs to know when the toolbar is open/closed and must have a properly aria-label (or something related) for each behavior and read it each time.

**Screenshots**
Schermata 2023-08-23 alle 16 36 09

**Software (please complete the following information):**

- Volto Version 17.0.0-alpha.19
- Plone Version 6.0.4

**Additional context**
There is some question to answer about this button: 
This button needs to be accessible by a screen-reader? 
Why a user using a screen reader need to shrink/expand this toolbar?

### PR DESCRIPTION
**Pull request by Noobham**
Improved toolbar accessibility. 
Hover effect [change in background color] on toolbar-handler.
Increased its contrast, and now the value for `aria-label` toggles correctly for both the expand and shrink statuses. 
fixes #5118 

https://github.com/plone/volto/assets/103531475/7b7f3eb8-2041-4639-b2c7-0f7a576c0200

### CODE REVIEW

**Comment by stevepiercy** [comment]
@Noobham this PR is incomplete and does not fix #5118 entirely. Please read the description again.

**Comment by stevepiercy** [comment]
@Noobham additionally do not close and open multiple PRs for the same issue. See item 3 in https://6.docs.plone.org/contributing/first-time.html#things-not-to-do

**Comment by Noobham** [comment]
@stevepiercy Thanks for reminding .
- Why these 3 unit test failing any idea about this??
- I have fixed accessibility problem .

**Comment by stevepiercy** [comment]
@Noobham have you read https://6.docs.plone.org/volto/contributing/testing.html and https://6.docs.plone.org/volto/contributing/acceptance-tests.html? Each of the failed CI jobs has a link to the explicit error message and why it fails. Example: https://github.com/plone/volto/actions/runs/6982100725/job/19000572605?pr=5431. However you should run these tests locally and get them to pass before pushing commits to the repo. This will make your development faster, save CI resources, and avoid sending notifications to subscribers. If you still get stuck, then you can push your work and ask for help, explaining what you specifically do not understand in the failure messages.

**Comment by Noobham** [comment]
> @Noobham have you read https://6.docs.plone.org/volto/contributing/testing.html and https://6.docs.plone.org/volto/contributing/acceptance-tests.html? Each of the failed CI jobs has a link to the explicit error message and why it fails. Example: https://github.com/plone/volto/actions/runs/6982100725/job/19000572605?pr=5431. However you should run these tests locally and get them to pass before pushing commits to the repo. This will make your development faster, save CI resources, and avoid sending notifications to subscribers. If you still get stuck, then you can push your work and ask for help, explaining what you specifically do not understand in the failure messages.

**Comment by Noobham** [comment]
@stevepiercy Thanks...

**Comment by Wagner3UB** [comment]
@ichim-david My first thought was to increase the contrast, but this will 'create' another color. In this case, wouldn't it be better to simply remove the opacity? What do you think?

**Comment by Noobham** [comment]
@ichim-david should i change color as it was before in this PR.

**Comment by ichim-david** [comment]
> @ichim-david should i change color as it was before in this PR.

@Noobham  I would do as I've said in my comment here https://github.com/plone/volto/pull/5431#issuecomment-1827477819
Unfortunately, I got no replies on this idea even when I asked for feedback. 
so:
1. set opacity to 0.65
2. set #e40166 color on hover

File: packages/volto/news/5118.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+fixed expand/shrink toolbar accessibility and ui of toolbar-handler - cms ui @Noobham
```
Comment (stevepiercy) [inline packages/volto/news/5118.bugfix:1]:
```suggestion
Improved toolbar accessibility. Increased its contrast, and now the value for `aria-label` toggles correctly for both the expand and shrink statuses. @Noobham
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This is clearer and explicit. Every bugfix fixes something, but it's better to say what was fixed.

File: packages/volto/news/5118.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Improved toolbar accessibility. Added hover effect on toolbar-handler, Increased its contrast, and now the value for `aria-label` toggles correctly for both the expand and shrink statuses. @Noobham
```
Comment (stevepiercy) [inline packages/volto/news/5118.bugfix:1]: Sorry, my previous review was sloppy. This is better.

```suggestion
Improved toolbar button accessibility by adding a hover effect and increasing its contrast, and now its value for `aria-label` toggles correctly for both its expand and shrink statuses. @Noobham
```

**Review by stevepiercy — CHANGES_REQUESTED** [review]

**Review by stevepiercy — APPROVED** [review]
I approve only the change log. @ichim-david @Wagner3UB would you please add your review of the change?

**Comment by ichim-david** [comment]
@Noobham it's holiday season so I don't know exactly who will take the time todo the technical review until next year.
There are 2 things that I would do extra:
1. Add aria-expanded property on the button https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded
2. Change the aria-label and affect also the sidebar button otherwise we would have a merge where the toolbar is all visible while the sidebar is with 0.3 opacity and with bad accessibility.
I consider both sections to be mirrors of each other so I would make both changes in same pull request.
 
Even then you might get more feedback from others but again be prepared to take into consideration that this pull request might be merged only in January as ppl get back to work from holiday season, including myself.

**Comment by Noobham** [comment]
@ichim-david What should I do with the sidebar button as I'm having trouble understanding the second section? Would you kindly ellaborate it please . 
Thanks.

**Comment by ichim-david** [comment]
> @ichim-david What should I do with the sidebar button as I'm having trouble understanding the second section? Would you kindly ellaborate it please . Thanks.

@Noobham 
add-content

You are logged in and you want to add a content or you edit one.
The toolbar and sidebar are both opened in edit.
The sidebar has similar button and functionality to the one found on the toolbar.

**Comment by Noobham** [comment]
https://github.com/plone/volto/assets/103531475/32acf93d-0b4e-46b8-a38f-0847ea94c1a3

Sidebar-handler's animation is playing when the page refreshes.
How should I approach animation?
@ichim-david

**Comment by ichim-david** [comment]
> Screen.Recording.2024-01-26.at.10.26.55.PM.mov
> 
> Sidebar-handler's animation is playing when the page refreshes. How should I approach animation? @ichim-david

@Noobham the color stays the same from light blue to a higher opacity blue, I didn't say for you to make the sidebar red, I said you need to apply the changes similar to what it was done in the toolbar.
"Change the aria-label and affect also the sidebar button otherwise we would have a merge where the toolbar is all visible while the sidebar is with 0.3 opacity and with bad accessibility."
Indeed I said that they were mirrors of each other but I thought you would have understood that I mean that they have the same functionality and style but they don't have the same color.

Regarding the animation that doesn't need to change, it is meant to bring awareness to the editor that it has those handles that can be interacted with.

**Comment by Noobham** [comment]
@ichim-david I apologise for not fully understanding what you said. 
from my understanding I need to adjust the sidebar's accessibility. but sidebar accessibility is already set.

**Comment by stevepiercy** [comment]
@Noobham as you said, you have already addressed the aria state toggle between open and closed, but I have not verified. I believe that @ichim-david wants only the opacity to be similar between the two button handles, for the toolbar and the sidebar, for their normal, hover, and clicked states. The color values would not change, and there would be no additional animation.

Also please note that you need to merge `main` into your branch and resolve the merge conflicts, as the GitHub UI in this issue indicates, then push your commits to your branch. See [Update your pull request from your fork](https://6.docs.plone.org/contributing/first-time.html#update-your-pull-request-from-your-fork).

**Comment by ichim-david** [comment]
Closing pull request as it's abandoned, last feedback was not taken into consideration and the logic for expanded is incorrect it should be inverted as seen from the test snapshot output.

**Pull request closed without merging**